Swipe to close support

Hello @ascoders. I want say thanks for that lib. 
Can you add "swipe to close" future?
I think that useful future.

@ascoders hi, i have some trouble with import local version react-native-image-pan-zoom

```console
$ rm -rf built && tsc
node_modules/@types/react-native/index.d.ts(8810,11): error TS2451: Cannot redeclare block-scoped variable 'global'.
```
What i do wrong?

package: 
`"react-native-image-pan-zoom": "file:../react-native-image-zoom/"`

P.S. local version image-viewer work nice with test app

--- a/readme.md
+++ b/readme.md
@@ -61,7 +61,7 @@ export default class App extends React.Component {
 |imageUrls|array|yes|Image Source||
 |enableImageZoom|boolean|no|Enable image zoom|`true`|
 |onShowModal|function<br><br>`(content?: JSX.Element) => void`|no|The callback for show modal|`() => {}`|
-|onCancel|funcion<br><br>`() => void`|no|The callback for cancel modal|`() => {}`|
+|onCancel|function<br><br>`() => void`|no|The callback for cancel modal|`() => {}`|
 |flipThreshold|number|no|Swipe threshold of the next page|`80`|
 |maxOverflow|number|no|The X position maximum, that current page can slide to the next page|`300`|
 |index|number|no|Init index of images|`0`|
@@ -78,6 +78,8 @@ export default class App extends React.Component {
 |renderIndicator|function<br><br>`(currentIndex?: number, allSize?) => React.ReactElement<any>`: number|no|Custom indicator|`(currentIndex, allSize) => currentIndex + "/" + allSize`|
 |renderArrowLeft|function<br><br>`() => React.ReactElement<any>`|no|Custom left arrow|`() => null`|
 |renderArrowRight|function<br><br>`() => React.ReactElement<any>`|no|Cutsom right arrow|`() => null`|
+|onSwipeDown|function<br><br>`() => void`|no|Callback for swipe down|`() => null`|
+|disableSwipeDown|boolean|no|Disable swipe down interactions|`false`|
 
 ## Development pattern
 

--- a/src/image-viewer.component.tsx
+++ b/src/image-viewer.component.tsx
@@ -91,7 +91,6 @@ export default class ImageViewer extends React.Component<Props, State> {
         status: "loading"
       })
     })
-
     // handle user swiping down
     const panResponderY = PanResponder.create({
       onStartShouldSetPanResponder: () => true,
@@ -112,7 +111,12 @@ export default class ImageViewer extends React.Component<Props, State> {
 
         if (y <= swipeDownThreshold) {
           // reset back to 0
-          this.positionY.setValue({ y: 0 })
+          Animated.timing(
+            this.positionY.y,
+            {
+              toValue: 0,
+            }
+          ).start();
         }
 
         if (y > swipeDownThreshold) {

--- a/src/image-viewer.component.tsx
+++ b/src/image-viewer.component.tsx
@@ -2,6 +2,7 @@ import * as React from "react"
 import {
   Animated,
   CameraRoll,
+  Dimensions,
   Image,
   Platform,
   Text,
@@ -9,6 +10,8 @@ import {
   TouchableOpacity,
   TouchableWithoutFeedback,
   View,
+  ViewStyle,
+  PanResponder
 } from "react-native"
 import ImageZoom from "react-native-image-pan-zoom"
 import styles from "./image-viewer.style"
@@ -27,6 +30,8 @@ export default class ImageViewer extends React.Component<Props, State> {
   // 整体位移，用来切换图片用
   private positionXNumber = 0
   private positionX = new Animated.Value(0)
+
+  private positionY = new Animated.ValueXY()
 
   private width = 0
   private height = 0
@@ -87,10 +92,42 @@ export default class ImageViewer extends React.Component<Props, State> {
       })
     })
 
+    // handle user swiping down
+    const panResponderY = PanResponder.create({
+      onStartShouldSetPanResponder: () => true,
+      onPanResponderMove: (event, gesture) => {
+        if (this.props.disableSwipeDown) {
+          return;
+        }
+
+        this.positionY.setValue({ y: gesture.dy })
+      },
+      onPanResponderRelease: (event) => {
+        if (this.props.disableSwipeDown) {
+          return;
+        }
+
+        const y = event.nativeEvent.pageY;
+        const swipeDownThreshold = 230 || this.props.swipeDownThreshold;
+
+        if (y <= swipeDownThreshold) {
+          // reset back to 0
+          this.positionY.setValue({ y: 0 })
+        }
+
+        if (y > swipeDownThreshold) {
+          // fire swipe down function
+          return this.props.onSwipeDown()
+        }
+      }
+    })
+
+
     this.setState(
       {
         currentShowIndex: nextProps.index,
-        imageSizes
+        imageSizes,
+        panResponderY,
       },
       () => {
         // 立刻预加载要看的图
@@ -578,49 +615,56 @@ export default class ImageViewer extends React.Component<Props, State> {
 
     return (
       <Animated.View
-        style={{ ...this.styles.container, opacity: this.fadeAnim }}
+        style={{ zIndex: 9999 }}
+        {...this.state.panResponderY.panHandlers}
+        style={this.positionY.getLayout()}
       >
-        {this!.props!.renderHeader!(this.state.currentShowIndex)}
-
-        <View style={this.styles.arrowLeftContainer}>
-          <TouchableWithoutFeedback onPress={this.goBack}>
-            <View>{this!.props!.renderArrowLeft!()}</View>
-          </TouchableWithoutFeedback>
-        </View>
-
-        <View style={this.styles.arrowRightContainer}>
-          <TouchableWithoutFeedback onPress={this.goNext}>
-            <View>{this!.props!.renderArrowRight!()}</View>
-          </TouchableWithoutFeedback>
-        </View>
-
         <Animated.View
-          style={{
-            ...this.styles.moveBox,
-            transform: [{ translateX: this.positionX }],
-            width: this.width * this.props.imageUrls.length
-          }}
+          style={{ ...this.styles.container, opacity: this.fadeAnim }}
         >
+          {this!.props!.renderHeader!(this.state.currentShowIndex)}
+
+          <View style={this.styles.arrowLeftContainer}>
+            <TouchableWithoutFeedback onPress={this.goBack}>
+              <View>{this!.props!.renderArrowLeft!()}</View>
+            </TouchableWithoutFeedback>
+          </View>
+
+          <View style={this.styles.arrowRightContainer}>
+            <TouchableWithoutFeedback onPress={this.goNext}>
+              <View>{this!.props!.renderArrowRight!()}</View>
+            </TouchableWithoutFeedback>
+          </View>
+
+          <Animated.View
+            style={{
+              ...this.styles.moveBox,
+              transform: [{ translateX: this.positionX }],
+              width: this.width * this.props.imageUrls.length
+            }}
+          >
           {ImageElements}
+
+          </Animated.View>
+
+          {
+            this!.props!.renderIndicator!(
+              (this.state.currentShowIndex || 0) + 1,
+              this.props.imageUrls.length
+            )
+          }
+
+          {this.props.imageUrls[this.state.currentShowIndex || 0].originSizeKb &&
+            this.props.imageUrls[this.state.currentShowIndex || 0].originUrl && (
+              <View style={this.styles.watchOrigin}>
+                <TouchableOpacity style={this.styles.watchOriginTouchable}>
+                  <Text style={this.styles.watchOriginText}>查看原图(2M)</Text>
+                </TouchableOpacity>
+              </View>
+            )}
+
+          {this!.props!.renderFooter!(this.state.currentShowIndex)}
         </Animated.View>
-
-        {
-          this!.props!.renderIndicator!(
-            (this.state.currentShowIndex || 0) + 1,
-            this.props.imageUrls.length
-          )
-        }
-
-        {this.props.imageUrls[this.state.currentShowIndex || 0].originSizeKb &&
-          this.props.imageUrls[this.state.currentShowIndex || 0].originUrl && (
-            <View style={this.styles.watchOrigin}>
-              <TouchableOpacity style={this.styles.watchOriginTouchable}>
-                <Text style={this.styles.watchOriginText}>查看原图(2M)</Text>
-              </TouchableOpacity>
-            </View>
-          )}
-
-        {this!.props!.renderFooter!(this.state.currentShowIndex)}
       </Animated.View>
     )
   }

--- a/src/image-viewer.type.ts
+++ b/src/image-viewer.type.ts
@@ -1,5 +1,5 @@
 import * as React from "react"
-import { Text, View, ViewStyle } from "react-native"
+import { ImageURISource, Text, View, ViewStyle } from "react-native"
 import { simpleStyle } from "./image-viewer.style"
 
 export class Props {
@@ -52,6 +52,16 @@ export class Props {
   public saveToLocalByLongPress?: boolean = true
 
   public style?: ViewStyle = {}
+
+  /**
+   * threshold for firing swipe down function
+   */
+  public swipeDownThreshold?: number = 230
+
+  /**
+   * for disabling vertical movement if user doens't want it
+   */
+  public disableSwipeDown?: boolean = false
 
   /**
    * 长按图片的回调
@@ -143,6 +153,13 @@ export class Props {
    * 取消看图的回调
    */
   public onCancel?: () => void = () => {
+    //
+  }
+
+  /**
+   * function that fires when user swipes down
+   */
+  public onSwipeDown?: () => void = () => {
     //
   }
 

